internal/resolution/manifest/maven.go

```diff
@@ -894,18 +895,19 @@ func updateString(enc *xml.Encoder, raw string, values map[string]string) error
 		if err != nil {
 			return err
 		}
-		if tt, ok := token.(xml.StartElement); ok {
+		if tt, ok := token.(xml.StartElement); ok && tt.Name.Local != tag {
+			var str string
+			if err := dec.DecodeElement(&str, &tt); err != nil {
+				return err
+			}
 			if value, ok2 := values[tt.Name.Local]; ok2 {
-				var str string
-				if err := dec.DecodeElement(&str, &tt); err != nil {
-					return err
-				}
-				if err := enc.EncodeElement(value, tt); err != nil {
-					return err
-				}
-
-				continue
+				str = value
+			}
+			if err := enc.EncodeElement(strings.TrimSpace(str), tt); err != nil {
```

Comment:
I spent some time trying to keep the white space but could not find a good way to do this, so I decided to trim out the white space. This is not ideal but good enough, @michaelkedar let me know how you feel about this.

Reply:
I'm fine with this if it's overly difficult to preserve the white space.
Though nit: maybe add a comment here explaining it.